server_fn/src/lib.rs

```diff
@@ -328,7 +329,23 @@ where
             // if redirected, call the redirect hook (if that's been set)
             if let Some(redirect_hook) = redirect_hook {
                 if (300..=399).contains(&status) || has_redirect_header {
-                    redirect_hook(&location);
+                    let origin = web_sys::window()
```

Comment:
The server function client is not guaranteed to have access to browser APIs, because server functions can also be used in desktop applications via `reqwest`, so this logic need to live somewhere in `leptos_router`